### PR TITLE
Feature/minio integration

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description

Added MinIO integration for doccument file persistence.

## How should this be tested?

Chose `document.file.storage=MINIOStorage` in application.properties. Try to save/edit/delete document files.

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: src/main/java/rs/teslaris/core/configuration/MinioClientConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Value("${spring.minio.access-key}")
+    private String minioAccessKey;
+
+    @Value("${spring.minio.secret-key}")
+    private String minioSecretKey;
+    
+    @Bean
+    public MinioClient minioClient() {
+        return MinioClient.builder().endpoint(minioHost).credentials(minioAccessKey, minioSecretKey)
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/configuration/MinioClientConfiguration.java:22]: This minioHost (url) also includes some name of collection/set? I suppose you can run one minio and use for different applications

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja I have added just two comments.

File: src/main/java/rs/teslaris/core/configuration/MinioClientConfiguration.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/configuration/MinioClientConfiguration.java:22]: Perhaps, I have not tried it.

File: docker-compose.yml [hunk 2]
```diff
@@ -9,16 +10,22 @@ services:
...
         environment:
             - API_PORT=8081
             - DB_USERNAME=postgres
             - DB_PASSWORD=postgres
             - DB_HOST=db
             - DB_PORT=5432
+            - MINIO_HOST=minio
+            - MINIO_BUCKET=teslaris-file-items
+            - MINIO_ACCESS_KEY=ROOTUSER
+            - MINIO_SECRET_KEY=CHANGEME123
```
[the remark this answers is not in the document]
Comment (ivanmrsulja) [reply docker-compose.yml:23]: it is provided to spring application, look at the application.properties:
```
spring.minio.url=http://${MINIO_HOST:localhost}:9000
spring.minio.bucket=${MINIO_BUCKET:teslaris-file-items}
spring.minio.access-key=${MINIO_ACCESS_KEY:ROOTUSER}
spring.minio.secret-key=${MINIO_SECRET_KEY:CHANGEME123}
```

[the remark this answers is not in the document]
Comment (chenejac) [reply docker-compose.yml:23]: If I am running the application out of docker, I should define these variables for maven (or in operating system), right? Should we add those variables in pom.xml?

File: src/main/java/rs/teslaris/core/configuration/MinioClientConfiguration.java (hunk 1 above)
Comment (chenejac) [reply src/main/java/rs/teslaris/core/configuration/MinioClientConfiguration.java:22]: Not sure I understand. MINIO_BUCKET is defined in environment variables, but not used by client. Maybe, the value defined in MINIO_BUCKET is applied by default. Anyway, I think we should investigate that to be fully under our control.

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja the merging conflicts should be resoled. I have also responded to your comments' responses.

**Review by chenejac — APPROVED** [review]

**Pull request merged**